### ISSUE TITLE
unify remote syntax for subdir across different remotes

### ISSUE THREAD

**Issue by kevinushey** [r-lib/remotes#526]
Currently, to install a package hosted on GitHub in a sub-directory, one would write:

```
remotes::install_github("user/repo/subdir")
```

However, this is not true for Gitlab, which supports sub-projects and so IIUC the sub-directory needs to be specified separately:

```
remotes::install_gitlab("group/project/subproject", subdir = "subdir")
```

I'd like to advocate unifying these syntaxes, so that:

- `user/repo/sub` always references a repository (or sub-repository);
- `user/repo:subdir` references the repository `user/repo`, but installation should proceed from subdir `subdir`.

In other words:

- Use `/` as a separator for user / repository fields.
- Use `:` as a delimiter for sub-directories within the selected user / repository.

**Comment by jimhester** [comment]
How would this handle backwards compatibility? I think there are a fair number of instances of people using subdirectories on GitHub.

Or would GitHub continue to use `/` for subdirectories, and only on GitLab would you need to use `:` to delineate the subdirectories?

**Comment by kevinushey** [comment]
I was thinking something like the following:

1. For GitHub only, support either `foo/bar/subdir` or `foo/bar:subdir` as a way of referencing a sub-directory (for backwards compatibility);

2. For all other remotes, support `foo/bar:subdir` as a way of referencing sub-directories in a repository, and `foo/bar/baz` as a way of referencing sub-projects;

3. If one day GitHub decided to support something similar (groups / projects / sub-projects), then the original syntax could be deprecated and moved towards the new form of declaring a sub-directory.

Does that sound reasonable? In other words, `:` would eventually become the "standard" delimiter for declaring a sub-directory, but `/` would be supported as an alternative for remotes which have no notion of sub-projects or sub-repositories.

**Comment by jimhester** [comment]
Conversely, is there another remote type that has this conflict other than GitLab?

**Comment by kevinushey** [comment]
It looks like Gitlab may be the only repository with this conflict.
